src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,572 @@
+            (1)
+            sage: E1.composition(E2) == E2.composition(E1)
+            True
+        """
+        p = []
+        n = []
+        z = []
+        for e in self.groundset():
+            x = self(e)
+            # If x is non-zero, keep its value
+            if x == 1:
+                p.append(e)
+            elif x == -1:
+                n.append(e)
+            else:
+                # else grab the value of the other
+                x = other(e)
+                if x == 1:
+                    p.append(e)
+                elif x == -1:
+                    n.append(e)
+                else:
+                    z.append(e)
+        return type(self)(self.parent(), positives=p, negatives=n, zeros=z)
+
+    def separation_set(self, other):
+        r"""
+        Return the separation set between two elements.
+
+        The separation set of two elements `X` and `Y`
+        is given by `S(X, Y) = \left\{e \mid X(e) = -Y(e) \neq 0 \right\}`
+        """
+        return self.positives().intersection(other.negatives()).union(self.negatives().intersection(other.positives()))
+
+    def reorientation(self, change_set):
+        r"""
+        Return the reorientation by a set.
+
+        The reorientation of `X` by some `A \subseteq E` is
+        the signed subset (covector) given by `{}_{-A}X` where
+        `{}_{-A}X^+ = (X^+ \backslash A) \cup (X^- \cap A)` and similarly for
+        `{}_{-A}X^-`.
+        """
+        if change_set in self.groundset():
+            change_set = set([change_set])
+        else:
+            change_set = set(change_set)
+
+        # ensure every elt is in the groundset
+        for i in change_set:
+            if i not in self.groundset():
+                raise ValueError(f"{i} is not in the groundset")
+
+        p = self.positives().difference(change_set).union(
+            self.negatives().intersection(change_set))
+        n = self.negatives().difference(change_set).union(
+            self.positives().intersection(change_set))
+        return type(self)(self.parent(), positives=p, negatives=n, groundset=self.groundset())
+
+    def is_conformal_with(self, other):
+        r"""
+        Return if the two elements are conformal.
+
+        Two elements `X` and `Y` are *conformal* if
+        `S(X,Y) = \emptyset`. This is true if and only if `X^+ \subseteq Y^+`
+        and `X^- \subseteq Y^-`.
+        """
+        return len(self.separation_set(other)) == 0
+
+    def is_restriction_of(self, other):
+        r"""
+        Return if `self` is a restriction of `other`.
+
+        A signed subset `X` is a *restriction* of a signed subset `Y` if
+        `X^+ \subseteq Y^+` and `X^- \subseteq Y^-`. If `X` is a restriction of
+        `Y` we sometimes say `X` conforms to `Y`. This should not be mistaken
+        with *is conformal with*.
+        """
+        return self.positives().issubset(other.positives()) \
+            and self.negatives().issubset(other.negatives())
```

Comment:
```suggestion
        return (self.positives().issubset(other.positives())
                and self.negatives().issubset(other.negatives()))
```
This is considered to be slightly better style due to less likely to introduce bugs.